submissions/NadiaVorontsova/classic-frogger-game/app.js

```diff
@@ -0,0 +1,136 @@
+const startPlayerPositionX = 200,
+    startPlayerPositionY = 390,
+    cellWidth = 101,
+    cellHeight = 83,
+    startX = 0,
+    startY = 0,
+    rightWall = 400,
+    bottomWall = 350;
+
+let countScore = 0;
+const scopeElement = document.createElement('div');
+scopeElement.style.cssText = 'font-size:25px; padding-top:20px; margin-bottom:-20px';
+document.body.appendChild(scopeElement);
+
+const Player = function (x, y) {
+    this.resetPlayerPosition();
+    this.sprite = 'images/char-boy.png';
+};
+
+Player.prototype.update = function () {
+    if (this.y < startY) {
+        this.resetPlayerPosition();
+        countScore++;
+        score();
+    }
+};
+
+Player.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Player.prototype.resetPlayerPosition = function () {
+    this.x = startPlayerPositionX;
+    this.y = startPlayerPositionY;
+}
+
+Player.prototype.handleInput = function (route) {
+    switch (route) {
+        case 'left':
+            if (this.x > startX)
+                this.x -= cellWidth;
+            break;
+        case 'up':
+            this.y -= cellHeight;
+            break;
+        case 'right':
+            if (this.x < rightWall)
+                this.x += cellWidth;
+            break;
+        case 'down':
+            if (this.y < bottomWall)
+                this.y += cellHeight;
+            break;
+        default:
+            break;
+    }
+};
+
+const player = new Player();
+
+const Enemy = function (x, y) {
+    this.x = x;
+    this.y = y;
+    this.speed = Math.random() * 200 + 80;
+    this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype.update = function (dt) {
+    this.x += this.speed * dt;
+    if (this.x > ctx.canvas.width) {
+        this.x = startX;
+    }
+    if (this.checkCollision()) {
+        player.resetPlayerPosition();
+    }
+};
+
+Enemy.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Enemy.prototype.checkCollision = function () {
+    if (Math.abs(this.x - player.x) < 50 && Math.abs(this.y - player.y) < 50) {
+        player.resetPlayerPosition();
+        if (countScore > 0) {
+            countScore--;
+            score();
+        }
+    }
+};
+
+const allEnemies = [];
+
+function createEnemies() {
+    const positionY = [60, 140, 220];
```

Comment:
As a peer developer I am tasked to add more rows to the field and more enemies to the game.
Can these mysterious numbers be calculated from something comprehensible?

Reply:
I changed this array